enhancements/ingress/gateway-api-crd-life-cycle-management.md

```diff
@@ -0,0 +1,467 @@
+> does not need to be carried forward to future releases as other logic exists
+> there to handle Gateway API CRD state (see below).
+
+### Post-upgrade
+
+> **Note**: The logic for these lives in in the new release (4.19) and onward.
+
+1. The CIO is hereafter deployed alongside its CRD protection VAP
+2. The CIO constantly checks for the presence of the CRDs
+  * IF the CRDs are present
+    * UNTIL the CRD schema matches what is expected, the CIO upgrades them
+    * IF the upgrade fails `Degraded` status is set
+  * ELSE the CRDs are deployed by the CIO
+
+> **Note**: If we reach `Degraded` its expected that some tampering has
+> occurred (e.g. a cluster-admin has for some reason destroyed our VAP and
+> manually changed the CRDs). For the initial release we will simply require
+> manual intervention (support) to fix this as we can't guess too well at the
+> original intent behind the change. In future iterations we may consider more
+> solutions if this becomes a common problem for some reason.
+
+### API Extensions
+
+None.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Hypershift runs the Ingress Operator on the management cluster but configured
+with a kubeconfig to manage resources on the guest cluster.  This means that the
+Ingress Operator manages the Gateway API CRDs on the guest cluster, the same as
+on standalone clusters.
+
+#### Standalone Clusters
+
+For standalone clusters, the Ingress Operator manages the Gateway API CRD
+life-cycle so that Gateway API can be configured and used post-install.
+
+#### Single-node Deployments or MicroShift
+
+The CRDs themselves use minimal resources.  Creating a GatewayClass CR can cause
+the Ingress Operator to install OpenShift Service Mesh, which in turn installs
+Istio and Envoy (see the [gateway-api-with-cluster-ingress-operator](gateway-api-with-cluster-ingress-operator.md)
+enhancement proposal), which use considerable resources.  For Single-Node
+OpenShift, the cluster-admin might be advised to pay particular attention to
+OSSM's resource requirements and the cluster's resource constraints before
+attempting to use Gateway API.
+
+MicroShift does not run the Ingress Operator and has its own design for
+supporting Gateway API (see the MicroShift [gateway-api-support](../microshift/gateway-api-support.md)
+enhancement).
+
+### Implementation Details/Notes/Constraints
+
+> What are some important details that didn't come across above in the
+> **Proposal**? Go in to as much detail as necessary here. This might be
+> a good place to talk about core concepts and how they relate. While it is useful
+> to go into the details of the code changes required, it is not necessary to show
+> how the code will be rewritten in the enhancement.
+
+### Risks and Mitigations
+
+#### Unknown Fields
+
+> **Note**: aka "dead fields"
+
+The definition of the "unknown fields" problem is provided upstream in
```

Comment:
Can you provide a brief explanation of what might happen for alternative implementations? It would be good to understand the risks our customers might face

Or do you think that's a bit scope creepy for this doc?

Reply:
https://github.com/kubernetes-sigs/gateway-api/issues/3624 (linked above) goes into more detail about the bad and worst cases that can arise from this problem, but I also updated the section above to be more clear that's where that information can be found. Let me know if you think that's sufficient ? If you prefer we talk more directly about it in this doc that's OK too.

Reply:
I agree with Joel, a brief explanation with a pointer to the upstream discussion would be great.

Reply:
I think what' you've added makes sense and helps, but giving a specific explanation of what might be seen might also help folks understand what impact dead fields might have, without having to explore the gateway-api issue linked, maybe something like:
```
For a third party implementation, that does not support the version of our Gateway API CRDs,
it may be possible for a user to specify a field, that has no effect immediately, but takes effect
once the implementation is later upgraded.
This could be confusing for end users, or could break their ingress during the upgrade, or it could even lead the user to incorrectly believe that their their endpoints are secure where the dead field is security related.
```
Just trying to hammer home why this is an important thing to consider

Reply:
```suggestion
The definition of the "unknown fields" problem is provided upstream in
For a third party implementation, that does not support the version of our Gateway API CRDs,
it may be possible for a user to specify a field, that has no effect immediately, but takes effect
once the implementation is later upgraded.
This could be confusing for end users, or could break their ingress during the upgrade, or it could even lead the user to incorrectly believe that their their endpoints are secure where the dead field is security related.
```

Reply:
78e5da5c9f40452147bd4daf35bc365e42df85ef includes a reworked version of the above which also includes some special notes about how we're going to document this for implementers. LMKWYT?